charm/kafka/reactive/kafka.py

```diff
@@ -114,22 +97,20 @@ def waiting_for_certificates():
 )
 @when_not('kafka.started')
 def configure_kafka(zk):
-    if len(zk.zookeepers()) < 1:
-        hookenv.status_set('blocked', 'waiting for >= 1 zookeeper units')
-        return
-
     hookenv.status_set('maintenance', 'setting up kafka')
+
     log_dir = unitdata.kv().get('kafka.storage.log_dir')
     data_changed('kafka.storage.log_dir', log_dir)
-    kafka = Kafka()
+
     zks = zk.zookeepers()
-    kafka.configure_kafka(zks, log_dir=log_dir)
+    kafka = Kafka()
+    kafka.install(zks=zks, log_dir=log_dir)
     kafka.open_ports()
+
     set_state('kafka.started')
+
     hookenv.status_set('active', 'ready')
-    # set app version string for juju status output
-    kafka_version = get_package_version(KAFKA_SNAP) or 'unknown'
-    hookenv.application_version_set(kafka_version)
+    hookenv.application_version_set(kafka.version())
```

Comment:
Should use snap layer's get_installed_version.

Reply:
@cmars kafka.version() calls get_installed_version.